### PR TITLE
[GLUTEN-CORE] Refactor cpp dependency on gtest and gbenchmark in more general way

### PR DESCRIPTION
**Pull request by Yohahaha**
Before this PR, gtest was installed at /usr/local/lib64 in our dev machine, and cpp/core/tests could not find gtest location to link by default.

After this PR, cmake will unify dependency management with FetchContent.

### CODE REVIEW

**Comment by Yohahaha** [comment]
@kerwin-zk

**Comment by Yohahaha** [comment]
CI failed due to oap-velox build error, seems already fixed.

```
/usr/bin/ld: ../../releases/libvelox.so: undefined reference to `facebook::velox::parquet::duckdb_reader::ParquetReader::ParquetReader(std::shared_ptr<facebook::velox::dwio::common::InputStream>, facebook::velox::dwio::common::ReaderOptions const&)'
```

**Comment by Yohahaha** [comment]
Hi @zhejiangxiaomai could you help take a look?

**Comment by zhejiangxiaomai** [comment]
> CI failed due to oap-velox build error, seems already fixed.
> 
> ```
> /usr/bin/ld: ../../releases/libvelox.so: undefined reference to `facebook::velox::parquet::duckdb_reader::ParquetReader::ParquetReader(std::shared_ptr<facebook::velox::dwio::common::InputStream>, facebook::velox::dwio::common::ReaderOptions const&)'
> ```

Thanks, I am fixing CI problem.

**Comment by zhejiangxiaomai** [comment]
I will merge it when CI pass.

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -105,6 +105,11 @@ find_package(Threads REQUIRED)
 find_package(JNI REQUIRED)
 
 if(BUILD_TESTS)
+  set(GLUTEN_GTEST_MIN_VERSION "1.12.1")
```
Comment (zhejiangxiaomai) [inline cpp/CMakeLists.txt:108]: Maybe 1.13.0 is more meaningful. We should keep gluten gtest version as same as velox.

Comment (Yohahaha) [reply cpp/CMakeLists.txt]: Thanks, already fix it.

**Review by zhejiangxiaomai — APPROVED** [review]

**Pull request merged**